src/ffi/index.ts

```
@@ -1,19 +1,159 @@
+) {
+  logger.warn(supportedPlatformsMessage);
+  logger.warn(detectedMessage);
+  logger.error(`Unsupported platform: ${platform}`);
+  throw new Error(
+    `Unsupported platform: ${platform}, set $UNSUPPORTED_PLATFORM to ${platform} to override this check`
+  );
+}
+
+if (process.env['UNSUPPORTED_PLATFORM'] !== platform) {
+  logger.warn(
+    `You have set the environment variable UNSUPPORTED_PLATFORM to ${process.env['UNSUPPORTED_PLATFORM']}, this will override the detected platform of ${platform}`
+  );
+}
+
+const loadPathMessage = (bindingsPath: string) =>
+  `: loading native module from: \n\n - ${path.join(
+    bindingsPath,
+    'prebuilds',
+    platform
+  )} ${
+    process.env['PACT_NAPI_NODE_LOCATION']
+      ? `\n - source: PACT_NAPI_NODE_LOCATION \n - You must have a supported prebuild for your platform at this location in the path ${path.join(
+          process.env['PACT_NAPI_NODE_LOCATION'],
+          'prebuilds',
+          platform
+        )}`
+      : `\n   source: ${path.join(
+          bindingsPath,
+          'prebuilds',
+          platform
+        )} \n\n - You can override via PACT_NAPI_NODE_LOCATION\n`
+  }`;
+
+const bindingsResolver = (bindingsPath: string | undefined) =>
+  bindings(bindingsPath);
+
+const bindingPaths = [
+  path.resolve(
+    process.env['PACT_NAPI_NODE_LOCATION']?.toString() ?? path.resolve(),
+    path.resolve(__dirname, '..', '..')
+  ),
+];
+let ffiLib: Ffi;
+let loadPath: string | undefined;
+try {
+  bindingPaths.forEach((bindingPath) => {
+    try {
+      loadPath = bindingPath;
+      logger.info(
+        `Attempting to find pact native module ${loadPathMessage(bindingPath)}`
+      );
+      ffiLib = bindingsResolver(bindingPath);
+      if (ffiLib) {
+        throw new Error('Native module not found');
+      }
+    } catch (error) {
+      logger.warn(`Failed to load native module from ${bindingPath}: ${error}`);
+    }
+  });
+} catch (error) {
```

This section isn't currently being caught, at least in the case that the `bindings` library can't find the node lib. The reason is that on line `90`, the error is caught there and thus is swallowed, both preventing the subsequent log from logging, and a non-zero exit status.